veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,166 @@
+# VEP #NNNN: NIC Hotplug / Hotunplug For Namespace Admins Following VMIM Hardening
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+Currently, the NIC hotplug / hotunplug feature depends on the user to manually initiate a migration following a 
+hotplug / hotunplug operation.
+The migration is required in order to attach / detach secondary interfaces from to / from the target virt-launcher pod.
+
+> [!NOTE]
+> SR-IOV hotplug operation requires a migration in order for the
+> [SR-IOV device plugin](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin) to be invoked when the
+> target
+> pod is created.
+
+[kubevirt/kubevirt#13497](https://github.com/kubevirt/kubevirt/pull/13497) had removed Namespace administrators'
+permissions to create VirtualMachineInstanceMigration objects - thus breaking the NIC hotplug / hotunplug feature
+for them in v1.5.
+
+## Motivation
+
+In previous KubeVirt versions, a project admin could hotplug / unplug NICs.
+We want to assure that this feature will continue working in v1.5 and future versions.
+
+## Goals
+
+Project admins could perform NIC hotplug / unplug operations without involving the cluster-admin for every hotplug /
+hotunplug operation.
+
+## Non Goals
+
+## Definition of Users
+
+- Namespace administrators
+
+## User Stories
+
+- As a namespace admin I would like to be able to hotplug / unplug a secondary interface without involving the cluster
+  admin.
+
+## Repos
+
+- kubevirt/kubevirt
+
+## Design
+
+### Integrating with LiveUpdate Rollout Strategy
+
+The CPU and memory hotplug features use the `LiveUpdate` `RolloutStrategy` which automatically initiates a migration
+of the VM when certain spec fields are changed.
+
+Extend the feature to automatically initiate a live migration in case of NIC hotplug / hotunplug request.
+
+#### Pros
+
+- Improves the current user experience
+- Will work for hotplugging interfaces using bridge and SR-IOV bindings
+
+#### Cons
+ - Will not work for non-migratable VMs
+ - Performs a migration for every hotplug / hotunplug
+ - Requires an API change
+ - Complicated implementation
+
+## API Examples
+
+<!--
+Tangible API examples used for discussion
+-->
+
+## Alternatives
+
+### Revert the VMIM Hardening PR
+
+[kubevirt/kubevirt#13497](https://github.com/kubevirt/kubevirt/pull/13497) will be reverted and namespace
+administrators could retain their permissions to create VMIM objects.
+
+#### Pros
+
+- Relatively simple
+- Allows for additional time to consider the preferred solution
+
+#### Cons
+
+- The VMIM hardening will be postponed to a later version
```

Comment:
Clusters are still exposed to an easy DoS on node drains and upgrades.

Reply:
> Now at least he is allowed to decide.

Yes, but a GA'ed functionality is broken for namespace admins.
Following the release of v1.5, cluster-admins will have to take action in order for namespace admins to be able to hotplug/hotunplug NICs.

Reply:
We also have a "dirty" workaround that requires no cluster-admin intervention:
do the first step of the hotplug as before, then instead of directly creating a VMIM object, hotplug a CPU core or a small amount of memory, this will implicitly trigger a live migration completing the nic hotplug.

Reply:
> instead of directly creating a VMIM object, hotplug a CPU core or a small amount of memory, this will implicitly trigger a live migration completing the nic hotplug.

Good luck documenting that :)
This is a no-go, we cannot do such hacks.